jme3-plugins-json-gson/src/main/java/com/jme3/plugins/gson/GsonArray.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2009-2023 jMonkeyEngine
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.plugins.gson;
+
+
+import java.util.Iterator;
+
+import com.jme3.plugins.json.*;
+
+/**
+ * GSON implementation of {@link JsonArray}.
+ */
+public class GsonArray extends GsonElement implements JsonArray {
+    public GsonArray(com.google.gson.JsonElement element) {
```

Comment:
This class should never be used directly and the documentation is already present in the interface it implements, copying the implementation here would be redundant and bad practice

Reply:
If it should never be used directly, then I think it should be `package-private`, if it cannot be, then it needs Javadoc to elaborate that it shouldn't be used and why it shouldn't, public classes without Javadoc will generate a lot of unwanted warnings while building the project.

Reply:
true, only GsonParser needs to be public, so i restricted the access to everything else, and since the module is included with implementation in gradle, it shouldn't case any leakage

Reply:
Thank you.